chia/full_node/hint_management.py

```diff
@@ -31,27 +29,23 @@ def add_if_ph_subscription(puzzle_hash: bytes32, coin_id: bytes32) -> None:
         if has_ph_subscription(puzzle_hash):
             lookup_coin_ids.add(coin_id)
 
-    for npc_result in state_change_summary.new_npc_results:
-        if npc_result.conds is None:
+    for spend_id, puzzle_hash in state_change_summary.removals:
+        # Record all coin_ids that we are interested in, that had changes
+        add_if_coin_subscription(spend_id)
+        add_if_ph_subscription(puzzle_hash, spend_id)
+
+    for addition_coin, hint in state_change_summary.additions:
+        addition_coin_name = addition_coin.name()
+        add_if_coin_subscription(addition_coin_name)
+        add_if_ph_subscription(addition_coin.puzzle_hash, addition_coin_name)
+        if hint is None:
             continue
-        for spend in npc_result.conds.spends:
-            # Record all coin_ids that we are interested in, that had changes
-            add_if_coin_subscription(bytes32(spend.coin_id))
-            add_if_ph_subscription(bytes32(spend.puzzle_hash), bytes32(spend.coin_id))
-
-            for new_ph, new_am, hint in spend.create_coin:
-                addition_coin: Coin = Coin(bytes32(spend.coin_id), bytes32(new_ph), uint64(new_am))
-                addition_coin_name = addition_coin.name()
-                add_if_coin_subscription(addition_coin_name)
-                add_if_ph_subscription(addition_coin.puzzle_hash, addition_coin_name)
-                if hint is None:
-                    continue
-                if len(hint) == 32:
-                    add_if_ph_subscription(bytes32(hint), addition_coin_name)
-
-                if len(hint) > 0:
-                    assert len(hint) <= 32
-                    hints_to_add.append((addition_coin_name, hint))
+        if len(hint) == 32:
+            add_if_ph_subscription(bytes32(hint), addition_coin_name)
+
+        if len(hint) > 0:
+            assert len(hint) <= 32
+            hints_to_add.append((addition_coin_name, hint))
```

Comment:
Things not exactly equal to length 32 are not considered hints right?

Reply:
they are, on the rust side: https://github.com/Chia-Network/chia_rs/blob/main/src/gen/conditions.rs#L346
since `bytes32` isn't available on the rust side, all are hinted as `bytes` anyway: https://github.com/Chia-Network/chia_rs/blob/main/wheel/chia_rs.pyi#L214